docs/guide/performance.md

```diff
@@ -0,0 +1,94 @@
+# Performance
+
+While Vite is fast by default, performance issues can creep in as the project's requirements grow. This guide aims to help you identify and fix common performance issues, such as:
+
+- Slow server starts
+- Slow page loads
+- Slow builds
+
+## Audit Configured Vite Plugins
+
+Vite's internal and official plugins are optimized to do the least amount of work possible. For example, code transformations use regex in dev, but do a complete parse in build to ensure correctness. However, the performance of community plugins is out of Vite's control, which may affect the developer experience.
```

Comment:
I would be less affirmative here. Some features were added for compatibility and sometimes reduce perf (like supporting stupid `.js` for TS files) or adding SaaS support.

I would prefer `Vite's internal and official plugins are optimized to do the least amount of work possible while providing compatibility with the broader ecosystem.`

Do you think we should add here some things like:
- Opt-out of SaaS if not extensively used or only used for CSS nesting
- Switch to the SWC plugin for React
- Use LightningCSS if possible

Reply:
I think those 3 points are slightly unrelated to this section, as the goal here is for users to check the community plugins they use.

But there could be a new section for the 3 points though, maybe "Use Lesser or Native Tooling". And we can suggest avoiding complexity and tooling for the written source code (e.g. use css instead of sass). And we can mention SWC and LightningCSS as the native tooling if they can afford the tradeoffs or larger install size.